timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionDetectionHelper.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    Set<PaxosValue> writeLogsOnServer(StateLogComponents server, int start, int end) {
+        return PaxosSerializationTestUtils.writeToLogs(server.acceptorLog(), server.learnerLog(), start, end);
+    }
+
+    void writeLogsOnLocalAndRemote(List<StateLogComponents> servers, int startingLogSeq, int latestLogSequence) {
+        servers.stream().forEach(server -> writeLogsOnServer(server, startingLogSeq, latestLogSequence));
+    }
+
+    StateLogComponents getDefaultLocalServer() {
+        return timeLockCorruptionTestSetup.getDefaultLocalServer();
+    }
+
+    List<StateLogComponents> getDefaultRemoteServerList() {
+        return timeLockCorruptionTestSetup.getDefaultRemoteServerList();
+    }
+
+    void induceGreaterAcceptedValueCorruptionOnDefaultLocalServer(int corruptSeq) {
+        induceGreaterAcceptedValueCorruption(timeLockCorruptionTestSetup.getDefaultLocalServer(), corruptSeq);
+    }
+
+    void induceGreaterAcceptedValueCorruption(StateLogComponents server, int corruptSeq) {
+        PaxosSerializationTestUtils.writeAcceptorStateForLogAndRound(
+                server.acceptorLog(),
+                corruptSeq,
+                Optional.of(PaxosSerializationTestUtils.createPaxosValueForRoundAndData(corruptSeq, corruptSeq + 1)));
+    }
+
+    List<CompletePaxosHistoryForNamespaceAndUseCase> getHistory() {
+        return timeLockCorruptionTestSetup.getPaxosLogHistoryProvider().getHistory();
+    }
+
+    void assertNoCorruptionViolations() {
+        assertViolationsDetectedForNamespaceAndUseCases(ImmutableSet.of(), ImmutableSet.of());
+    }
+
+    void assertAcceptedValueGreaterThanLearnedValue() {
+        assertViolationsDetectedForNamespaceAndUseCases(
+                ImmutableSet.of(CorruptionCheckViolation.ACCEPTED_VALUE_GREATER_THAN_LEARNED),
+                ImmutableSet.of(Constants.DEFAULT_NAMESPACE_AND_USE_CASE));
+    }
+
+    void assertViolationsDetected(Set<CorruptionCheckViolation> detectedViolations) {
+        assertViolationsDetectedForNamespaceAndUseCases(
+                detectedViolations, ImmutableSet.of(Constants.DEFAULT_NAMESPACE_AND_USE_CASE));
+    }
+
+    void assertViolationsDetectedForNamespaceAndUseCases(
+            Set<CorruptionCheckViolation> detectedViolations,
+            Set<NamespaceAndUseCase> namespaceAndUseCasesWithViolation) {
+
+        Multimap<CorruptionCheckViolation, NamespaceAndUseCase> violationsToNamespaceToUseCaseMultimap =
+                getViolationsToNamespaceToUseCaseMultimap();
+
+        assertThat(violationsToNamespaceToUseCaseMultimap.keySet()).hasSameElementsAs(detectedViolations);
+        assertThat(violationsToNamespaceToUseCaseMultimap.values())
+                .hasSameElementsAs(namespaceAndUseCasesWithViolation);
+    }
+
+    private Multimap<CorruptionCheckViolation, NamespaceAndUseCase> getViolationsToNamespaceToUseCaseMultimap() {
+        List<CompletePaxosHistoryForNamespaceAndUseCase> historyForAll = getHistory();
```

Comment:
you can inline this variable - the method call is actually shorter than the variable!